File: app/src/main/java/com/adityap/flashy_createflashcards/models/FlashcardModel.java
package com.adityap.flashy_createflashcards.models;

/**
 * Created by Aditya on 9/30/2018.
 */

public class FlashcardModel {
    private int mId;
    private String mWord;
    private String mDefinition;


    public  FlashcardModel(){
        mWord = null;
        mDefinition = null;
    }

    public FlashcardModel(String word, String definition){
        mWord = word;
        mDefinition = definition;
    }

    public int getmId() {
        return mId;
    }

    public void setmId(int mId) {
        this.mId = mId;
    }
    public String getWord() {
        return mWord;
    }

    public void setWord(String mWord) {
        this.mWord = mWord;
    }

    public String getDefinition() {
        return mDefinition;
    }

    public void setDefinition(String mDefinition) {
        this.mDefinition = mDefinition;
    }
}

File: app/src/main/java/com/adityap/flashy_createflashcards/models/FlashcardModel.kt
package com.adityap.flashy_createflashcards.models

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by Aditya on 9/30/2018.
 */
data class FlashcardModel(val id: Int = 0, val word: String, val definition: String) : Parcelable {
    constructor(parcel: Parcel) : this(
            parcel.readInt(),
            requireNotNull(parcel.readString()),
            requireNotNull(parcel.readString()))

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeInt(id)
        parcel.writeString(word)
        parcel.writeString(definition)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<FlashcardModel> {
        override fun createFromParcel(parcel: Parcel): FlashcardModel {
            return FlashcardModel(parcel)
        }

        override fun newArray(size: Int): Array<FlashcardModel?> {
            return arrayOfNulls(size)
        }
    }

    fun toCardPair(): CardPair =
            CardPair(front = CardModel(title = this.word),
                    back = CardModel(description = this.definition))

}
